networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py

```diff
@@ -211,7 +227,14 @@ def _query(self, resource_type, key, ands=[], ors=[], dsl="",
                size=50, start=0):
         data = self._get_query(resource_type, key, ands, ors, dsl, size, start)
         resp = self._post(path="/nsxapi/rpc/call/SearchFacade", data=data)
-        return json.loads(resp.content)["result"]["results"]
+        content = json.loads(resp.content)
+
+        if content.get("result") is None\
+                or content.get("result").get("results") is None:
+            raise HttpUnsuccessfulExceptoin(
+                "Expected 'results' in payload={}".format(resp.content))
+
+        return content.get("result").get("results")
```

Comment:
Why the `get` here? Doesn't the check 4 lines up cover this?

Reply:
Yes, the above check covers this.